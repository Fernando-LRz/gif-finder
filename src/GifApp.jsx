import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

const GifApp = () => {

    const [ categories, setCategories ] = useState([]);

    const onAddCategory = (newCategory) => {
        if( categories.includes( newCategory ) ) return;

        // setCategories(category => [ ...category, 'D' ]);
        setCategories([ ...categories, newCategory ]);
    }; 

    return (
        <>
            <h1>GIF finder</h1>

            <AddCategory 
                // setCategories={ setCategories }
                // onNewCategory={ (value) => onAddCategory(value) }
                onNewCategory={ onAddCategory }
            />

            {
                categories.map(( category ) => (
                    <GifGrid key={ category } category={ category }/>
                ))
            }

        </>
    );
};

export default GifApp;