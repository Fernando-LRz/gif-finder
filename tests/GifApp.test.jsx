import { fireEvent, render, screen } from '@testing-library/react';
import GifApp from '../src/GifApp';

describe('Pruebas en <GifApp />', () => {

    const title = 'GifApp';
    const category = 'JavaScript';

    test('Hacer match con el snapshot', () => {
        
        const { container } = render( <GifApp /> );
        expect( container ).toMatchSnapshot();
        
    });

    test('Mostrar el título en un h1', () => {

        render( <GifApp /> );

        const h1 = screen.getByRole('heading', { level: 1 });
        
        expect( h1.innerHTML ).toBe(title);
        // screen.debug();

    });

    test('Evitar la misma categoria múltiples veces', () => {

        render( <GifApp /> );

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form );

        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form );

        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form );

        expect( screen.getAllByRole('heading', { level: 3, value: category }).length ).toBe(1);

    });


});