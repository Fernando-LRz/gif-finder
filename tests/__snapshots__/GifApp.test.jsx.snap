// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> Hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
